### PR TITLE
resolve expression without assignment warnings

### PR DESCRIPTION
**Pull request by jenningsje**

### CODE REVIEW

File: src/sos/__main__.py [hunk 1]
```diff
@@ -912,7 +912,6 @@ def cmd_worker(args, workflow_args):
         from .workers import SoS_Worker
 
         procs = [SoS_Worker(env.config) for i in range(args.workers)]
-        [p.start() for p in procs]
```
Comment (BoPeng) [inline src/sos/__main__.py:915]: How can you remove the statement? As I explained to you before, this expression has the side effect of starting all the `p`. To remove the warning, you can change it to 

```
for p in proces:
    p.start()
```

File: src/sos/section_analyzer.py [hunk 2]
```diff
@@ -356,7 +355,7 @@ def get_step_depends(section):
                     "STEP",
                     f"Args {value} in depends cannot be determined: {e}")
         finally:
-            [env.sos_dict.dict().pop(x) for x in svars]
+            old_values = [env.sos_dict.dict().pop(x) for x in svars]
```
Comment (BoPeng) [inline src/sos/section_analyzer.py:358]: Again, `old_values` is useful variable. Please change this expression to a for loop.

File: src/sos/section_analyzer.py [hunk 3]
```diff
@@ -409,7 +408,7 @@ def get_step_input(section, default_input):
         # expression ...
         step_input = sos_targets(_undetermined=stmt)
     finally:
-        [env.sos_dict.dict().pop(x) for x in svars]
+        old_values = [env.sos_dict.dict().pop(x) for x in svars]
```
Comment (BoPeng) [inline src/sos/section_analyzer.py:411]: The same, the expression was used for its side effect (`.pop()`), so change to a for loop.

File: src/sos/step_executor.py [hunk 4]
```diff
@@ -129,13 +129,6 @@ def get_job(self, all_tasks=False):
         # individual tasks...
         if self.trunk_size == 1 or all_tasks:
             to_be_submitted = self._unsubmitted_tasks
-            [
-                to_be_submitted.extend([x[1]
-                                        for x in slot
-                                        if x[1] is not None])
-                for slot in self._slots
-                if slot
-            ]
```
Comment (BoPeng) [inline src/sos/step_executor.py:138]: You cannot simply remove a statement.... Change to a for loop if you understand this piece of code.

**Comment by jenningsje** [comment]
I'm making these corrections right now.

File: src/sos/workers.py [hunk 5]
```diff
@@ -757,4 +757,4 @@ def kill_all(self):
             total_num_workers -= 1
             self.report(f"Kill {msg[1:]}")
         # join all local processes
-        [worker.join() for worker in self._local_workers]
+
```
Comment (bioworkflows) [inline src/sos/workers.py:760]: You cannot just remove the entire expression.

File: src/sos/targets.py [hunk 6]
```diff
@@ -1103,7 +1103,6 @@ def __init__(
             self._handle_for_each(for_each)
 
     def set_traced(self):
-        [x.set_traced() for x in self._targets]
```
Comment (bioworkflows) [inline src/sos/targets.py:1106]: you cannot just remove the expression.

File: src/sos/section_analyzer.py [hunk 7]
```diff
@@ -535,7 +536,7 @@ def get_step_output(section, default_output, analysis_type):
                     f'Failed to determine input "{value}" of an auxiliary step: {e}'
                 ) from e
         finally:
-            [env.sos_dict.dict().pop(x) for x in svars]
+            old_values = [env.sos_dict.dict().pop(x) for x in svars]
```
Comment (bioworkflows) [inline src/sos/section_analyzer.py:539]: You cannot change the meaning of the variable `old_values`.

File: .github/linters/.python-lint [hunk 8]
```diff
@@ -84,11 +84,11 @@ disable=abstract-method,
       comprehension-escape,
       consider-iterating-dictionary,
       consider-using-dict-items,
-      consider-using-f-string,
       consider-using-generator,
       consider-using-get,
       consider-using-in,
       consider-using-set-comprehension,
+      consider-using-f-string,
```
Comment (bioworkflows) [inline .github/linters/.python-lint:91]: Please put the warning in the right place.

File: src/sos/targets.py [hunk 9]
```diff
@@ -1103,9 +1103,8 @@ def __init__(
...
-        self.traced = True
```
Comment (BoPeng) [inline src/sos/targets.py:1107]: why did you remove these two lines?

File: src/sos/workers.py [hunk 10]
```diff
@@ -737,7 +737,7 @@ def check_workers(self):
             self._local_worker_alive_time = time.time()
             # join processes if they are now gone, it should not do anything bad
             # if the process is still running
-            [
+            self._num_workers = [
```
Comment (BoPeng) [inline src/sos/workers.py:740]: This is not fixed. You cannot just assign the return value to a used variable.

File: src/sos/__main__.py [hunk 11]
```diff
@@ -140,12 +140,14 @@ def get_convert_parser(desc_only=False):
             subparser = add_sub_parser(
                 subparsers,
                 converter_parser,
-                name="{}-{}".format(f_format, t_format))
+                name=f"{f_format}-{t_format}"
```
Comment (BoPeng) [inline src/sos/__main__.py:143]: This change belongs to another PR. Please revert.

File: src/sos/__main__.py [hunk 12]
```diff
@@ -140,12 +140,14 @@ def get_convert_parser(desc_only=False):
...
+            )
             subparser.add_argument(
                 "from_file",
                 metavar="FROM",
                 nargs="?",
-                help="""File to be converted.""")
+                help="""File to be converted."""
+            )
```
Comment (BoPeng) [inline src/sos/__main__.py:150]: This as well. You might get merge issues if you do not revert.

**Review by BoPeng — APPROVED** [review]
